server/src/main/java/org/elasticsearch/action/admin/indices/diskusage/IndexDiskUsageAnalyzer.java

```diff
@@ -526,6 +532,38 @@ void visitField(Fields vectors, String fieldName) throws IOException {
         }
     }
 
+    void analyzeKnnVectors(SegmentReader reader, IndexDiskUsageStats stats) throws IOException {
+        KnnVectorsReader vectorReader = reader.getVectorReader();
+        if (vectorReader == null) {
+            return;
+        }
+        for (FieldInfo field : reader.getFieldInfos()) {
+            cancellationChecker.checkForCancellation();
+            directory.resetBytesRead();
+            if (field.getVectorDimension() > 0) {
+                iterateDocValues(reader.maxDoc(), () -> vectorReader.getVectorValues(field.name), vectors -> {
+                    cancellationChecker.logEvent();
+                    vectors.vectorValue();
+                });
+
+                // do a couple of randomized searches to figure out min and max offsets of index file
+                VectorValues vectorValues = vectorReader.getVectorValues(field.name);
+                int numDocsToVisit = reader.maxDoc() < 1000 ? reader.maxDoc() : 1000 * (int) Math.log10(reader.maxDoc());
+                int skipFactor = reader.maxDoc() / numDocsToVisit;
+                for (int i = 0; i < reader.maxDoc(); i += skipFactor) {
+                    if ((i = vectorValues.advance(i)) == DocIdSetIterator.NO_MORE_DOCS) {
+                        break;
+                    }
+                    cancellationChecker.checkForCancellation();
+                    vectorReader.search(field.name, vectorValues.vectorValue(), 1, null, 20);
```

Comment:
A `visitedLimit` of 20 is quite low and might mean that you only end up visiting the upper layers to the graph over and over again, instead of exploring the bottom layer (the full nearest neighbor graph). Also setting `k=1` means the search could search ends really early.

Maybe something like `k=1000, visitedLimit=1000` would make this more stable? Or some function of the number of docs?